mapiproxy/servers/default/emsmdb/oxcfxics.c

```diff
@@ -3353,8 +3403,27 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopSyncImportReadStateChanges(TALLOC_CTX *mem_c
 
 	request = &mapi_req->u.mapi_SyncImportReadStateChanges;
 
+	retval = openchangedb_get_MailboxReplica(emsmdbp_ctx->oc_ctx, emsmdbp_get_owner(synccontext_object), NULL, &replica_guid);
+	if (retval != MAPI_E_SUCCESS) {
+		OC_DEBUG(5, "Impossible to get %s mailbox replica guid", emsmdbp_get_owner(synccontext_object));
+		mapi_repl->error_code = MAPI_E_CALL_FAILED;
```

Comment:
You can test it without a problem following these steps:

* Modify this code to return always MAPI_E_CALL_FAILED
* Mark a message as unread
* Sync the folder

According to spec, it will try it using Online mode (SetMessageReadFlag ROP), so no sync issues are expected. Indeed, in Outlook 2010 it tries both at the same time.